pingcap/dm
# DM-worker: refine lock usage

### What problem does this PR solve? 

1. get lock only if needed
2. deep copy some protobuf message

### What is changed and how it works?

1. only get the lock when access `subTasks` in `Worker. handleTask`
2. change `Lock` to `RLock` in `Worker.Status`
3. use `proto.Clone` to deep copy a protobuf message

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
 	1. modify Load/Flush source code for checkpoint to sleep some time
 	2. use `start-task`/`stop-task` to operate the subtask (these commands will return with RPC timeout)
 	3. use `query-status` to query the status (should without RPC timeout)
